src/WebJobs.Script.WebHost/Diagnostics/DiagnosticListenerService.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public sealed class DiagnosticListenerService : IHostedService, IDisposable
+    {
+        private readonly ILogger<DiagnosticListenerService> _logger;
+        private readonly Dictionary<string, IDisposable> _listenerSubscriptions;
+        private readonly IDisposable _standbyChangeHandler;
+        private static bool _debugTraceEnabled;
+        private IDisposable _allListenersSubscription;
+        private bool _disposed;
+
+        public DiagnosticListenerService(ILogger<DiagnosticListenerService> logger, IOptionsMonitor<StandbyOptions> standbyOptions)
+        {
+            _logger = logger;
+            _listenerSubscriptions = new Dictionary<string, IDisposable>();
+
+            if (standbyOptions.CurrentValue.InStandbyMode)
+            {
+                _standbyChangeHandler = standbyOptions.OnChange(o => SubscribeListeners());
```

Comment:
For new listeners it would, but if they're setup on something that won't change (e.g., the assembly load context), it wouldn't, but I see why you're asking, as when I initially implemented this, the logic enabled full sources (source names) to be for debug only, which means we'd not subscribe to them in placeholder mode, but that was moved to the event name, so this wouldn't be necessary.

I'll still need to handle the change, but only to flip the debug flag. Good catch!